libmscore/transpose.cpp

```diff
@@ -509,9 +509,7 @@ void Score::transposeKeys(int staffStart, int staffEnd, int tickStart, int tickE
                               segmentInterval.flip();
                         }
                   KeySig* ks = toKeySig(s->element(staffIdx * VOICES));
-                  if (!ks)
-                        continue;
-                  if (ks->generated())
+                  if (!ks || (ks->generated() && ks->segment()->rtick() > 0))
```

Comment:
@jthistle - why the added check on rtick()?  This is measure-relative tick meaning it only triggers for generated mid-measure key changes.  Pretty sure that's not a thing.  Removing this fixes https://musescore.org/en/node/280370, but I'm still trying to figure out if it was needed to fix the issue in your PR.  It appears originally didn't have that, then you added it when combining the !ks and ks->generated() checks.